src/TimeModeling/judiWeights.jl

```diff
@@ -41,26 +43,37 @@ can also be a single (non-cell) array, in which case the weights are the same fo
     judiWeights(weights; nsrc=1)
 """
 function judiWeights(weights::Array; nsrc=1, vDT::DataType=Float32)
-    vDT == Float32 || throw(judiWeightsException("Domain type not supported"))
-
+    (eltype(weights) != vDT) && (weights = convert(Array{vDT},weights))
     # length of vector
     n = 1
     m = prod(size(weights))*nsrc
     weightsCell = Array{Array}(undef, nsrc)
     for j=1:nsrc
         weightsCell[j] = deepcopy(weights)
     end
-    return judiWeights{Float32}("Extended source weights",m,n,nsrc,weightsCell)
+    return judiWeights{vDT}("Extended source weights",m,n,nsrc,weightsCell)
 end
 
 # constructor if weights are passed as a cell array
 function judiWeights(weights::Union{Array{Any,1},Array{Array,1}}; vDT::DataType=Float32)
-    vDT == Float32 || throw(judiWeightsException("Domain and range types not supported"))
+    nsrc = length(weights)
+    for i = 1:nsrc
+        (eltype(weights[i]) != vDT) && (weights[i] = convert(Array{vDT},weights[i]))
+    end
+    # length of vector
+    n = 1
+    m = prod(size(weights[1]))*nsrc
+    return judiWeights{vDT}("Extended source weights",m,n,nsrc,weights)
+end
+
+#
+function judiWeights(weights::Array{Array{iDT, N},1}; vDT::DataType=Float32) where {iDT,N}
```

Comment:
why a separate constructor? This should be with above.

Reply:
This one is separated because we could check the type match very fast -- we know its type by iDT. We only have to check once (iDT == vDT) to decide whether to convert types. In Array{Array,1} and Array{Any, 1}, the only way to check type match is to loop over all shots and check individually. So I make them separate. Is this OK?